src/plugin.ts

```diff
@@ -65,8 +65,13 @@ function activateWidgetExtension(app: JupyterLab, registry: IJupyterWidgetRegist
               v._outputView.activate();
             });
           } else {
-            app.shell.add(w, 'right');
-            app.shell.expandRight();
+            let anchor = this.model.get('anchor');
+            if(anchor === 'right'){
```

Comment:
if you allow this attribute to be None, then this is backwards compatible. Otherwise, the js below will have an error with an older version of sidecar where the attribute is not defined.

How about making the attribute allowed to be None (i.e., optional), and handling that case here as 'right' here as well?

Reply:
In other words, adding an attribute to the data model is fine as long as that attribute is completely optional. In other words, this js code now will not work if communicating with an older model.

So I think if you just handle the case where the model does not have an `anchor` attribute, we're set - then it will handle working with an older version of the python code.

Reply:
Thanks for the latest suggestion Jason. I committed the needed changes to make `anchor` optional.